src/main/java/utils/HttpRequest.java

```
@@ -0,0 +1,152 @@
+            break;
+        case PUT:
+            request = new HttpPut();
+            break;
+        }
+    }
+
+    /*
+     * Function that clean parameters field only.
+     */
+    public void cleanParams() {
+        params.clear();
+    }
+
+    /*
+     * Function that resets the HTTP Request object so it can be reused.
+     *
+     */
+    public void resetHttpRequestObject() {
+        this.cleanParams();
+        request.reset();
+    }
+
+    /*
+     * Function for adding headers to the http request.
+     *
+     * @param key , the key of the header
+     * @param value, the value of the header
+     *
+     * @return HTTPRequest
+     */
+    public HttpRequest addHeader(String key, String value) {
+        request.addHeader(key, value);
+        return this;
+    }
+
+    /*
+     * Function for adding parameters to the http request.
+     *
+     * @param key , the key of the parameter
+     * @param value, the value of the parameter
+     *
+     * @return HTTPRequest
+     */
+    public HttpRequest addParam(String key, String value) {
+        params.put(key, value);
+        return this;
+    }
+
+    /*
+     * Function that set the body of the http request.
+     *
+     * @param body , the body to be set in the http request.
+     *
+     * @return HTTPRequest
+     */
+    public HttpRequest setBody(String body) {
+        ((HttpEntityEnclosingRequestBase) request).setEntity(new StringEntity(body, "UTF-8"));
+        return this;
+    }
+
+    /*
+     * Function that set the body of the http request.
+     *
+     * @param body , the file with body content to be set in the http request.
+     */
+    public void setBody(File file) {
+        String fileContent = "";
+        try {
+            fileContent = FileUtils.readFileToString(file, "UTF-8");
+        } catch (IOException e) {
+            LOGGER.error("Failed to read the Request body file:" + file.getPath() + ". Message: " + e.getMessage(), e);
```

Should this maybe throw an error? It feels like if this goes wrong, the code should not continue?